lib/pages/media_player/media_playback_screen.dart

```diff
@@ -0,0 +1,388 @@
+import 'package:flutter/material.dart';
+import '../../constants/constants.dart';
+import '../playlist/song_details_screen.dart';
+import '../../widgets/media_player/video_player.dart';
+import 'package:just_audio/just_audio.dart';
+import 'package:audio_session/audio_session.dart';
+import 'lyrics_section.dart';
+
+class MediaPlayback extends StatefulWidget {
+  const MediaPlayback({super.key});
+
+  @override
+  State<MediaPlayback> createState() => _MediaPlaybackState();
+}
+
+class _MediaPlaybackState extends State<MediaPlayback> {
+  bool _isLiked = false;
+  final AudioPlayer _audioPlayer = AudioPlayer();
+  bool _isPlaying = false;
+
+  @override
+  void initState() {
+    super.initState();
+    // Initialize audio
+    _initAudio();
+  }
+
+  Future<void> _initAudio() async {
+    try {
+      // Remove AudioSession configuration
+      await _audioPlayer.setAudioSource(
+        AudioSource.asset('assets/audio/Where I come from - Passion Pit.mp3'),
+        initialPosition: Duration.zero,
+        preload: true,
+      );
+
+      // Set audio mixing behavior
+      await _audioPlayer.setVolume(1.0);
+      await _audioPlayer.setAndroidAudioAttributes(
+        const AndroidAudioAttributes(
+          contentType: AndroidAudioContentType.music,
+          usage: AndroidAudioUsage.media,
+          flags: AndroidAudioFlags.none,
+        ),
+      );
+
+      // Listen for player state changes
+      _audioPlayer.playerStateStream.listen((state) {
+        if (mounted) {
+          setState(() {
+            _isPlaying = state.playing;
+          });
+        }
+      });
+
+      // Add this: Listen for playback completion
+      _audioPlayer.processingStateStream.listen((state) {
+        if (state == ProcessingState.completed) {
+          setState(() {
+            _isPlaying = false;
+          });
+          _audioPlayer.seek(Duration.zero);  // Reset to beginning
+        }
+      });
+      
+    } catch (e) {
+      print('Error initializing audio: $e');
+    }
+  }
+
+  void _handlePlayPause() async {
+    try {
+      if (_isPlaying) {
+        await _audioPlayer.pause();
+      } else {
+        await _audioPlayer.play();
+      }
+    } catch (e) {
+      print('Error handling play/pause: $e');
+    }
+  }
+
+  String _formatDuration(Duration duration) {
```

Comment:
it is advised to create a functions folder for all functions. Pass parameters if necessary